account_statement_so_completion/models/account_move.py

```diff
@@ -60,27 +58,14 @@ def get_from_ref_and_so(self, cr, uid, st_line, context=None):
 
             ...}
         """
-        st_obj = self.pool['account.bank.statement.line']
         res = {}
-        if st_line:
-            so_obj = self.pool.get('sale.order')
-            so_id = so_obj.search(
-                cr, uid, [('name', '=', st_line['ref'])], context=context)
-            if so_id:
-                if so_id and len(so_id) == 1:
-                    so = so_obj.browse(cr, uid, so_id[0], context=context)
-                    res['partner_id'] = so.partner_id.id
-                elif so_id and len(so_id) > 1:
-                    raise ErrorTooManyPartner(
-                        _('Line named "%s" (Ref:%s) was matched by more '
-                          'than one partner while looking on SO by ref.') %
-                        (st_line['name'], st_line['ref']))
-                st_vals = st_obj.get_values_for_line(
-                    cr, uid, profile_id=st_line['profile_id'],
-                    master_account_id=st_line['master_account_id'],
-                    partner_id=res.get('partner_id', False),
-                    line_type='customer',
-                    amount=st_line['amount'] if st_line['amount'] else 0.0,
-                    context=context)
-                res.update(st_vals)
+        so_obj = self.env['sale.order']
+        orders = so_obj.search([('name', '=', line.name)])
+        if len(orders) > 1:
+            raise ErrorTooManyPartner(
```

Comment:
This exception seems wrong. Here we have too many sale orders not partners. Maybe we simply want a `self.ensure_one()` in such case.